WooCommerce/Classes/ViewRelated/Products/Edit Product/Product Settings/ProductSettingsSections.swift

```diff
@@ -21,19 +21,14 @@ enum ProductSettingsSections {
         let rows: [ProductSettingsRowMediator]
 
         init(_ settings: ProductSettings) {
-            if settings.productType == .simple {
-                let tempRows: [ProductSettingsRowMediator?] = [ProductSettingsRows.Status(settings),
-                        ProductSettingsRows.Visibility(settings),
-                        ProductSettingsRows.CatalogVisibility(settings),
-                        ProductSettingsRows.VirtualProduct(settings),
-                        ProductSettingsRows.DownloadableProduct(settings)
-                ]
-                rows = tempRows.compactMap { $0 }
-            } else {
-                rows = [ProductSettingsRows.Status(settings),
-                        ProductSettingsRows.Visibility(settings),
-                        ProductSettingsRows.CatalogVisibility(settings)]
-            }
+            let shouldShowVirtualProductSetting = settings.productType == .simple
+            let shouldShowDownloadableProductSetting = settings.productType == .simple || settings.productType == .subscription
+            let rows: [ProductSettingsRowMediator?] = [ProductSettingsRows.Status(settings),
+                    ProductSettingsRows.Visibility(settings),
+                    ProductSettingsRows.CatalogVisibility(settings),
+                    shouldShowVirtualProductSetting ? ProductSettingsRows.VirtualProduct(settings) : nil,
+                                                      shouldShowDownloadableProductSetting ? ProductSettingsRows.DownloadableProduct(settings) : nil]
```

Comment:
just a np but this line seems to have too many whitespace?

I like the logic simplification here, though!

Reply:
Fixed indentation in 8d9407b72467709d312a747592159086e7fd732d